### PR TITLE
docs(inputs): Improve focus documentation

### PR DESCRIPTION
**Pull request by dimodi**

### CODE REVIEW

File: knowledge-base/inputs-set-focus.md [hunk 1]
```diff
@@ -6,88 +6,140 @@ page_title: Focus textbox or input programmatically
...
+        await DropDownListRef.FocusAsync();
     }
 }
+````
+
+## Focus on Page Load
+
+* The **earliest reliable programmatic focus** can occur in `OnAfterRenderAsync` and with some delay. The reason is that `OnAfterRenderAsync` is fired when the DOM tree is built, but **before** the HTML output is actually rendered in the browser. After the event is fired, the .NET runtime sends the HTML to the browser.
+The `FocusAsync` method relies on `JSInterop`, which in turn relies on the component to be rendered in the browser.
+* You can still use JavaScript to focus DOM elements by having a proper element reference or selector. The C# method is built on top of that. If you want more specific functionality (like <a href="https://feedback.telerik.com/blazor/1454982-always-highlight-all-numerictextbox-content-on-focus" target="_blank">selecting the text</a> as well, a pure JS solution might be simpler).
```
Comment (yordan-mitev) [inline knowledge-base/inputs-set-focus.md:100]:
```suggestion
* You can still use JavaScript to focus DOM elements by having a proper element reference or selector. The C# method is built on top of that. If you want more specific functionality (like <a href="https://feedback.telerik.com/blazor/1454982-always-highlight-all-numerictextbox-content-on-focus" target="_blank">selecting the text</a> as well), a pure JS solution might be simpler.
```

File: components/editor/overview.md [hunk 2]
```diff
@@ -99,11 +99,14 @@ The following table lists Editor parameters, which are not discussed elsewhere i
 
 ## Editor Reference and Methods
 
-You can use the component reference to call its `ExecuteAsync` method, especially when creating [custom tools]({%slug editor-custom-tools%}). This method allows you to call programmatically the tools and commands of the editor (such as the Bold too, or a Back Color tool, or inserting HTML).
+The Editor provides methods for programmatic operation. To use them, obtain reference to the component via its `@ref` attribute.
```
Comment (yordan-mitev) [inline components/editor/overview.md:102]:
```suggestion
The Editor provides methods for programmatic operation. To use them, obtain a reference to the component through its `@ref` attribute.
```

File: components/maskedtextbox/overview.md [hunk 3]
```diff
@@ -92,7 +91,7 @@ The following parameters enable you to customize the [appearance]({%slug maskedt
 
 ## MaskedTextBox Reference and Methods
 
-The MaskedTextBox proves a `FocusAsync` method that enables programmatic focus. To use it, obtain reference to the component instance.
+The MaskedTextBox proves a `FocusAsync` method that enables programmatic focus. To use it, obtain reference to the component instance via `#ref`. @[template](/_contentTemplates/common/inputs.md#focus-kb)
```
Comment (yordan-mitev) [inline components/maskedtextbox/overview.md:94]: This sentence appears in the rest of the `overview.md` files, I suggest changing it there as well.
```suggestion
The MaskedTextBox proves a `FocusAsync` method that enables programmatic focus. To use it, obtain a reference to the component instance through `#ref`. @[template](/_contentTemplates/common/inputs.md#focus-kb)
```

**Review by yordan-mitev — APPROVED** [review]

File: components/numerictextbox/overview.md [hunk 4]
```diff
@@ -111,7 +111,7 @@ The following parameters enable you to customize the [appearance]({%slug numeric
 
 ## Numeric TextBox Reference and Methods
 
-The Numeric TextBox has a `FocusAsync` method that enables programmatic focus. To use it, obtain a reference to the component instance.
+The Numeric TextBox has a `FocusAsync` method that enables programmatic focus. To use it, obtain a reference to the component instance via `@ref`. @[template](/_contentTemplates/common/inputs.md#focus-kb)
```
Comment (dimodi) [inline components/numerictextbox/overview.md:114]:
```suggestion
The Numeric TextBox has a `FocusAsync` method that enables programmatic focus. To use it, obtain a reference to the component instance through `@ref`. @[template](/_contentTemplates/common/inputs.md#focus-kb)
```

File: components/radiogroup/overview.md [hunk 5]
```diff
@@ -101,6 +101,47 @@ The Blazor RadioGroup provides various parameters to configure the component. Al
 
 See the [Input Validation]({%slug common-features/input-validation%}) article for more details.
 
+## RadioGroup Reference and Methods
+
+The RadioGroup provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component via its `@ref` attribute.
```
Comment (dimodi) [inline components/radiogroup/overview.md:106]:
```suggestion
The RadioGroup provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component through its `@ref` attribute.
```

File: components/textarea/overview.md [hunk 6]
```diff
@@ -71,6 +71,32 @@ The Blazor TextArea provides various parameters to configure the component:
 
 See the [Input Validation]({%slug common-features/input-validation%}) article.
 
+## TextArea Reference and Methods
+
+The TextArea provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component via its `@ref` attribute.
```
Comment (dimodi) [inline components/textarea/overview.md:76]:
```suggestion
The TextArea provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component through its `@ref` attribute.
```

File: components/textbox/overview.md [hunk 7]
```diff
@@ -67,6 +67,32 @@ The Blazor TextBox provides various parameters to configure the component:
 
 See also the [Input Validation]({%slug common-features/input-validation%}) article.
 
+## TextBox Reference and Methods
+
+The TextBox provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component via its `@ref` attribute.
```
Comment (dimodi) [inline components/textbox/overview.md:72]:
```suggestion
The TextBox provides a `FocusAsync` method that allows the application to focus the component programmatically. First, obtain reference to the component through its `@ref` attribute.
```

File: knowledge-base/inputs-set-focus.md [hunk 8]
```diff
@@ -6,88 +6,140 @@ page_title: Focus textbox or input programmatically
...
     {
-        await DropdownRef.FocusAsync();
+        await DropDownListRef.FocusAsync();
     }
 }
+````
+
+## Focus on Page Load
+
+* The **earliest reliable programmatic focus** can occur in `OnAfterRenderAsync` and with some delay. The reason is that `OnAfterRenderAsync` is fired when the DOM tree is built, but **before** the HTML output is actually rendered in the browser. After the event is fired, the .NET runtime sends the HTML to the browser.
```
Comment (dimodi) [inline knowledge-base/inputs-set-focus.md:98]:
```suggestion
* The earliest reliable programmatic focus can occur in `OnAfterRenderAsync` and with some delay. The reason is that `OnAfterRenderAsync` is fired when the DOM tree is built, but before the HTML output is actually rendered in the browser. After the event is fired, the .NET runtime sends the HTML to the browser.
```

**Pull request merged**